### PR TITLE
feat: refactor atomic components to functional

### LINKED ISSUE #1761: [FEATURE] Convert atomic components into functional ones
**Describe how this feature will help us improve**
Describe why do you think this feature is needed. 

Functional components are smaller and lighter which result in better performance. Considering that many atomic components (and possibly some molecules and organisms) are view only (or does not implement complex functionality) they can be converted into functional ones.

**Is your proposal related to a problem?**
Describe the issue or type its number

**Describe the solution you'd considered**
A concise description of your preferred solution. Things to address include:

https://medium.com/js-dojo/vue-js-functional-components-what-why-and-when-439cfaa08713

  - Details of the technical implementation
  - Tradeoffs made in design decisions
  - Caveats and considerations for the future

If there are multiple solutions, please present each one separately. Save comparisons for the very end.

**Additional context**
Anything you would like to add but was not sure which section it fits.

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue
Closes #1761

Scope of work
Detect components which can be refectored to functional and process refactoring:

- SfBreadcrumbs
- SfBullet
- SfButton - adding focus directive to be resolved
- SfColor
- SfHeading
- SfIcon
- SfLink - adding focus directive to be resolved
- SfQuantitySelector
- SfRating
- SfSkeleton
- SfTextarea

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed
- [x] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js [hunk 1]
```diff
@@ -20,7 +20,12 @@ export default {
         category: "Props",
       },
     },
-    click: { action: "set Value of current", table: { category: "Events" } },
+    // click: { action: "set Value of current", table: { category: "Events" } },
```
Comment (justyna-13) [inline packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js:23]: do we need it here? :)

File: packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js [hunk 2]
```diff
@@ -20,7 +20,12 @@ export default {
...
+  },
+  parameters: {
+    actions: {
+      handles: ['click', '.sf-bullet']
```
Comment (justyna-13) [inline packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js:27]: should be there any info about this click in the actions tab? I can't see anything :)

**Review by justyna-13 — APPROVED** [review]
Good job!

File: packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js (hunk 1 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js:23]: Of course not, thanks :)

File: packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js (hunk 2 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/atoms/SfBullets/SfBullets.stories.js:27]: I tried to implement it to make them work and show in action tab at the same time, but it doesn't work. I remove it.

**Review by Baroshem — APPROVED** [review]
Cool!

**Pull request merged**